src/main/java/nurseybook/model/task/UniqueTaskList.java

```diff
@@ -126,7 +126,8 @@ public void updateElderlyNameInTasks(Elderly target, Elderly editedElderly) {
         for (Task task : internalList) {
             for (Name name : task.getRelatedNames()) {
                 if (target.getName().caseInsensitiveEquals(name)) {
-                    task.replaceName(name, editedElderly.getName());
+                    internalList.remove(task);
```

Comment:
This is fine since list is sorted with every task command so the order will be fixed.

Reply:
Wait nevermind, I'm changing this to get index then set. Though deleteElderly should not be done in task view, if it was executed, this implementation would mess up the tasklist.